probe-rs-tools/src/bin/probe-rs/cmd/info.rs

```diff
@@ -440,21 +450,64 @@ fn coresight_component_tree(
     Ok(tree)
 }
 
+fn process_known_rom_tables(
+    interface: &mut dyn ArmProbeInterface,
+    id: &ComponentId,
+    _table: &RomTable,
+    access_port: MemoryAp,
+    tree: &mut Tree<String>,
+) -> Result<()> {
+    let peripheral_id = id.peripheral_id();
+    let Some(part_info) = peripheral_id.determine_part() else {
+        return Ok(());
+    };
+
+    if part_info.peripheral_type() == PeripheralType::Custom && part_info.name() == "Atmel DSU" {
```

Comment:
This is the kind of thing I do not want :/ Special casing per chip in random places. It will deteriorate as quick as OpenOCD :/
Do we have any plans to make this more generic? I am not against special casing (we need it) but we should have a system for it where one can add any number of cases without littering the code. Similar to let's say a service router (not exactly like that ofc :D).

Reply:
> Do we have any plans to make this more generic?

This isn't per chip, this is per manufacturer :) I hear your concern and we'll see what I can do here. I have plans for chip identification but that idea isn't very well integrated into the "parse random ROM tables and print their data" idea of `info`.

Reply:
Ok :) I am fine with having this for now, but let's put emphasis on cleaning it up rather soon :)